File: 06_손민기/1주차/햄버거 만들기.py
def solution(ingredient):
    # 1 = 빵, 2 = 야채, 3 = 고기
    hambuger = [1,2,3,1]
    stack = []
    answer = 0
    
    for i in range(len(ingredient)):
        stack.append(ingredient[i])
    
        # 마지막 idx와 길이를 비교하는 조건을 통해 시간단축을 할 수 있다
        if stack[-1] == hambuger[-1] and len(stack) >= len(hambuger):
            if stack[-len(hambuger):] == hambuger:
                stack[-len(hambuger):] = []
                answer += 1
            
    return answer

File: 햄버거만들기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> ingredient) { 
    int answer = 0;
    int i = 0;
 
   while(1){

        if (i == ingredient.size() - 3 || ingredient.size() < 4) break;

       else if (ingredient[i] == 1 && ingredient[i + 1] == 2 && ingredient[i + 2] == 3 && ingredient[i + 3] == 1) {

           for (int k = 0; k < 4; k++) {
               ingredient.erase(ingredient.begin() + i);
           }
           answer++;
           i = 0;
       }
     
       else i++;
   }

    return answer;
}
